manager/apis/app/v1alpha1/plotter_types.go

```diff
@@ -4,21 +4,223 @@
+	// +required
+	Steps [][]DataFlowStep `json:"steps" protobuf:"bytes,11,opt,name=steps"`
+}
+
+// Flows is the list of data flows driven from fybrikapplication:
+// Each element in the list holds the flow of the data requested in fybrikapplication.
+type Flow struct {
+	// Name of the flow
+	// +required
+	Name string `json:"name"`
+
+	// Type of the flow (e.g. read)
+	// +required
+	FlowType DataFlow `json:"flowType"`
+
+	// AssetID indicates the data set being used in this data flow
+	// +required
+	AssetID string `json:"assetId"`
+
+	// +required
+	SubFlows []SubFlow `json:"subFlows"`
+}
+
+// ModuleInfo is a copy of M4DModule Custom Resource.  It contains information
+// to instantiate resource of type M4DModule.
+type ModuleInfo struct {
+	// Name of the module
+	// +required
+	Name string `json:"name"`
+
+	// May be one of service, config or plugin
+	// Service: Means that the control plane deploys the component that performs the capability
+	// Config: Another pre-installed service performs the capability and the module deployed configures it for the particular workload or dataset
+	// Plugin: Indicates that this module performs a capability as part of another service or module rather than as a stand-alone module
+	// +required
+	Type string `json:"type"`
+
+	// Kind of k8s resource
+	// +required
+	Kind string `json:"kind"`
+
+	// Chart contains the information needed to use helm to install the capability
+	// +required
+	Chart ChartSpec `json:"chart"`
+
+	// Capabilities declares what this module knows how to do and the types of data it knows how to handle
+	// The key to the map is a CapabilityType string
+	// +required
+	Capabilities []ModuleCapability `json:"capabilities"`
+}
+
+// Template contains basic information about the required modules to serve the fybrikapplication
+// e.g., the module helm chart name.
+type Template struct {
+	// Name of the template
+	// +required
+	Name string `json:"name,omitempty"`
+
+	// Modules is a list of dependent modules. e.g., if a plugin module is used
+	// then the service module is used in should appear first in the modules list of the
+	// same template. If the modules list contains more than one module, the first module in the list
+	// is referred to as the "primary module" of which all the parameters to this template are
+	// sent to.
+	// +required
+	Modules []ModuleInfo `json:"modules"`
+}
+
+// FlowStatus includes information to be reported back to the FybrikApplication resource
+// It holds the status per data flow
+type FlowStatus struct {
+	// ObservedState includes information about the current flow
+	// It includes readiness and error indications, as well as user instructions
+	// +optional
+	ObservedState ObservedState `json:"status,omitempty"`
+
+	// +required
+	SubFlows map[string]ObservedState `json:"subFlows"`
+}
+
 // PlotterSpec defines the desired state of Plotter, which is applied in a multi-clustered environment. Plotter installs the runtime environment
```

Comment:
Small technicality, but the plotter doesn't install the runtime environment.  It declares what needs to be installed and where?

Reply:
The comment is updated.